tests/shadow/test_shadow_measurement.py

```diff
@@ -0,0 +1,240 @@
+        dev = qml.device("default.qubit", wires=wires, shots=shots)
+        res = qml.classical_shadow(wires=range(wires), seed_recipes=seed)
+        assert res.shape(device=dev) == (2, shots, wires)
+
+        # test an error is raised when device is None
+        msg = "The device argument is required to obtain the shape of a classical shadow measurement process"
+        with pytest.raises(qml.measurements.MeasurementShapeError, match=msg):
+            res.shape(device=None)
+
+    @pytest.mark.parametrize("wires", wires_list)
+    @pytest.mark.parametrize("shots", shots_list)
+    @pytest.mark.parametrize("seed", seed_recipes_list)
+    def test_measurement_process_copy(self, wires, shots, seed):
+        """Test that the attributes of the MeasurementProcess instance are
+        correctly copied"""
+        dev = qml.device("default.qubit", wires=wires, shots=shots)
+        res = qml.classical_shadow(wires=range(wires), seed_recipes=seed)
+
+        copied_res = copy.copy(res)
+        assert type(copied_res) == type(res)
+        assert copied_res.return_type == res.return_type
+        assert copied_res.wires == res.wires
+        assert copied_res.seed == res.seed
+
+    @pytest.mark.all_interfaces
+    @pytest.mark.parametrize("wires", wires_list)
+    @pytest.mark.parametrize("shots", shots_list)
+    @pytest.mark.parametrize("seed", seed_recipes_list)
+    @pytest.mark.parametrize("interface", ["autograd", "jax", "tf", "torch"])
+    @pytest.mark.parametrize("device", ["default.qubit", "default.mixed", None])
+    def test_format(self, wires, shots, seed, interface, device):
+        """Test that the format of the returned classical shadow
+        measurement is correct"""
+        import torch
+
+        circuit = get_circuit(wires, shots, seed, interface, device)
+        shadow = circuit()
+
+        # test shape and dtype are correct
+        assert shadow.shape == (2, shots, wires)
+        assert shadow.dtype == np.uint8 if interface != "torch" else torch.uint8
+
+        bits, recipes = shadow
+
+        # test allowed values of bits and recipes
+        assert qml.math.all(np.logical_or(bits == 0, bits == 1))
+        assert qml.math.all(np.logical_or(recipes == 0, np.logical_or(recipes == 1, recipes == 2)))
+
+    @pytest.mark.all_interfaces
+    @pytest.mark.parametrize("wires", wires_list)
+    @pytest.mark.parametrize("interface", ["autograd", "jax", "tf", "torch"])
+    @pytest.mark.parametrize("device", ["default.qubit", None])
+    @pytest.mark.parametrize(
+        "circuit_fn, basis_recipe",
+        [(get_x_basis_circuit, 0), (get_y_basis_circuit, 1), (get_z_basis_circuit, 2)],
+    )
+    def test_return_distribution(self, wires, interface, device, circuit_fn, basis_recipe):
+        """Test that the distribution of the bits and recipes are correct for a circuit
+        that prepares all qubits in a Pauli basis"""
+        # high number of shots to prevent true negatives
+        shots = 5000
```

Comment:
for smaller qubit number this can potentially also be reduced

Reply:
It should stay the same. Actually it can only be reduced for larger numbers of qubits, since more samples means less likely to deviate too far from the average.

Reply:
that sounds counterintuitive 🤔 can you elaborate?

Reply:
I want to test if the distribution of 0s and 1s is roughly 50-50. The more samples I have, the closer the distribution will be to the ideal (in the limit). More qubits means more samples, so if there are less qubits, there must be more samples to make up for it.